daemon/src/model/cfd.rs

```diff
@@ -2272,6 +2290,52 @@ mod tests {
         assert_eq!(funding_fee.rate, funding_rate);
     }
 
+    #[test]
+    fn given_negative_funding_rate_then_long_margin() {
+        let price = Price::new(dec!(10000)).unwrap();
+        let quantity = Usd::new(dec!(200));
+        let leverage = Leverage::new(2).unwrap();
+
+        let funding_rate = dec!(-1);
+
+        let cfd = Cfd::dummy_not_open_yet()
+            .with_position(Position::Long)
+            .with_margin(price, quantity, leverage);
+
+        let margin = cfd.margin_for_funding_rate(FundingRate::new(funding_rate).unwrap());
+        assert_eq!(margin, Amount::from_btc(0.01).unwrap());
```

Comment:
I'd call unwrap on the operation (`margin_for_funding_rate`) and not in the assertion.